// @ryanshaug/globallistings-test-stamp-duty-calculator/karma.conf.js
var webpackCfg = require('./webpack.config');
webpackCfg.devtool = 'inline-source-map';

webpackCfg.module.loaders[0].query.plugins = ['transform-runtime'];

module.exports = function(config) {
    config.set({
        basePath: '',
        browsers: ['PhantomJS'],
        port: 8080,
        captureTimeout: 60000,
        frameworks: ['jasmine'],
        client: {
            mocha: {}
        },
        singleRun: true,
        files: [
            './node_modules/phantomjs-polyfill/bind-polyfill.js',
            'tests.webpack.js'
        ],

        preprocessors: {
            'tests.webpack.js': ['webpack', 'sourcemap']
        },
        webpack: webpackCfg,
        webpackServer: {
            noInfo: true
        },
        externals: {
            'react/addons': true,
            'react/lib/ExecutionEnvironment': true,
            'react/lib/ReactContext': true
        },
        resolve: {
            modules: ['src', 'node_modules'],
            extensions: ['', '.js', '.jsx']
        }
    });
};
